src/cloudWatchLogs/cloudWatchLogsUtils.ts

```diff
@@ -120,4 +120,26 @@ export function findOccurencesOf(document: vscode.TextDocument, keyword: string)
     return ranges
 }
 
+const HIGHLIGHTER = vscode.window.createTextEditorDecorationType({
+    backgroundColor: new vscode.ThemeColor('list.focusHighlightForeground'),
+})
+
+export async function highlightDocument(registry: LogStreamRegistry, uri: vscode.Uri): Promise<void> {
+    const textEditor = registry.getTextEditor(uri)
+    const logData = registry.getLogData(uri)
+
+    if (!logData) {
+        throw new Error(`Missing log data in registry for uri key: ${uriToKey(uri)}. Unable to highlight`)
```

Comment:
todo/future pr: should this use ToolkitError?

Reply:
I didn't know this existed but after looking into it, is there any reason not to use ToolkitError for errors within the toolkit?

Reply:
It's on-going work but ideally most contributors would just use `ToolkitError`. 

The message you use might be shown directly to the user so it needs a little more thought. A normal `Error` won't be treated the same as we don't know where the error came from. 

Right now, your current message would be confusing to a user but potentially helpful for a Toolkit developer. We'd want to phrase the message differently and put useful data somewhere else in a structured way:
```ts
throw new ToolkitError(`Failed to highlight document "${uri.path}". No log data found.`, { details: { key: uriToKey(uri) } })
```